spec/src/main/asciidoc/core/spi_lite.asciidoc

```diff
@@ -1,12 +1,386 @@
 [[spi_lite]]
 
-== Build time extensions
+== Build compatible extensions
 
-// TODO
-[NOTE]
-====
-TBD
-====
+A build compatible extension may integrate with the container during deployment time, as defined in <<initialization>>.
 
-// Must mention that Build Compatible Extensions do not have access to the CDI container,
-// as there may not be one running! Calling `CDI.current()` inside them results in non-portable behavior.
+Comprehensive specification of build compatible extensions is available in the javadoc.
+This text provides an overview.
+
+[[bce]]
+
+=== The `BuildCompatibleExtension` interface
+
+A build compatible extension is a service provider of the `jakarta.enterprise.inject.build.compatible.spi.BuildCompatibleExtension` interface, declared in `META-INF/services`.
+
+[source, java]
+----
+public interface BuildCompatibleExtension {}
+----
+
+Build compatible extensions can define arbitrary `public`, non-`static`, `void`-returning methods without type parameters, annotated with one of the extension annotations.
+Such methods are called _extension methods_.
+
+Extension annotations correspond to extension execution phases:
```

Comment:
```suggestion
Extension method annotations correspond to extension execution phases:
```

Reply:
The previous paragraph introduces the term "extension annotation". I'm not entirely happy about introducing so many terms, but I can't think of a better way.

Reply:
Not done, to avoid breaking the flow of the text.